internal/asyncapi/account_details.go

```diff
@@ -52,43 +52,40 @@ func (d *accountDetails) getTags() error {
 		return catchOpenAPIError(err)
 	}
 	for _, topicLevelTag := range topicLevelTags {
-		d.channelDetails.topicLevelTags = append(d.channelDetails.topicLevelTags, spec.Tag{Name: topicLevelTag.TypeName})
+		d.channelDetails.topicLevelTags = append(d.channelDetails.topicLevelTags, spec.Tag{Name: topicLevelTag.GetTypeName()})
 	}
 
 	// Get schema level tags
 	d.channelDetails.schemaLevelTags = nil
-	schemaLevelTags, err := d.srClient.GetTags("sr_schema", strconv.Itoa(int(d.channelDetails.schema.Id)))
+	schemaLevelTags, err := d.srClient.GetTags("sr_schema", strconv.Itoa(int(d.channelDetails.schema.GetId())))
 	if err != nil {
 		return catchOpenAPIError(err)
 	}
 	for _, schemaLevelTag := range schemaLevelTags {
-		d.channelDetails.schemaLevelTags = append(d.channelDetails.schemaLevelTags, spec.Tag{Name: schemaLevelTag.TypeName})
+		d.channelDetails.schemaLevelTags = append(d.channelDetails.schemaLevelTags, spec.Tag{Name: schemaLevelTag.GetTypeName()})
 	}
 	return nil
 }
 
 func (d *accountDetails) getSchemaDetails() error {
-	schema, err := d.srClient.GetSchemaByVersion(d.channelDetails.currentSubject, "latest", nil)
+	schema, err := d.srClient.GetSchemaByVersion(d.channelDetails.currentSubject, "latest", false)
 	if err != nil {
 		return err
 	}
 	d.channelDetails.schema = &schema
 
 	// The backend considers "AVRO" to be the default schema type.
-	if schema.SchemaType == "" {
-		schema.SchemaType = "AVRO"
-	}
-
-	if schema.SchemaType == "PROTOBUF" {
+	switch schema.GetSchemaType() {
+	case "":
+		schema.SchemaType = srsdk.PtrString("AVRO")
+	case "PROTOBUF":
 		return fmt.Errorf("protobuf is not supported")
+	case "AVRO", "JSON":
```

Comment:
Ah wait this doesn't work... The `""` case has to be moved outside of the `switch`

Reply:
oops